creation/views.py

```diff
@@ -2861,3 +2943,567 @@ def update_codefiles(request):
+        
+    # Pagination
+    paginator = Paginator(tutorials, 50)
+    try:
+        page = int(request.GET.get('page','1'))
+    except:
+        page = int(request.GET.get('page'))
+
+    try:
+        posts = paginator.page(page)
+        
+    except(EmptyPage,PageNotAnInteger):
+        posts = paginator.page(paginator.num_pages)
+    context['collection'] = posts
+    context['header'] = header
+    context['ordering'] = ordering
+    context['status'] = active
+    context['counter'] = itertools.count(1)
+
+    if request.method == 'POST':
+        language_id = request.POST.get('language')
+        if language_id:
+            contributors_updated = RoleRequest.objects.filter(role_type=0,status = 1,language=language_id).values_list('user__id').distinct();
+            rated_contributors = ContributorRating.objects.filter(language_id=language_id,user_id__in=contributors_updated).values_list('user_id','user__username','rating')
+            context['contributors'] = rated_contributors
+    
+    context.update(csrf(request))
+    if is_language_manager(request.user):
+        return render(request, 'creation/templates/allocate_tutorial_manager.html', context)
+    else:
+        return render(request, 'creation/templates/allocate_tutorial.html', context)
+
+#global variable
+submissiondate = datetime.date(datetime.now())
+
+def refresh_tutorials(request):
+    count = 0
+    tutorials = TutorialResource.objects.filter(script_status =4,language = 22)
+    contrib_langs = RoleRequest.objects.filter(status=1,user=request.user).values_list('language')
+    for tutorial in tutorials:
+        this_tut_langs = Language.objects.exclude(id__in= TutorialResource.objects.filter(
+                    tutorial_detail_id = tutorial.tutorial_detail.id).values('language'))
+        
+        sam = Language.objects.filter(id__in = this_tut_langs).exclude(Q(id=22)&Q(id__in=contrib_langs))
+        for a_lang in sam:
+            already_present = TutorialsAvailable.objects.filter(tutorial_detail_id=tutorial.tutorial_detail.id,language=a_lang).exists()
+            if not already_present:
+                tutorialsavailable =  TutorialsAvailable()
+                tutorialsavailable.tutorial_detail = tutorial.tutorial_detail
+                tutorialsavailable.language =  a_lang
+                count +=1
+                tutorialsavailable.save()
+
+    if count>0:
+        messages.success(request,str(count)+' tutorials added')
+    else:
+        messages.warning(request, "No tutorials updated")
+
+    return HttpResponseRedirect(request.META['HTTP_REFERER'])
+
+def no_of_foss_gt_3(user):
+    """Check if the user is having domain reviewer rights"""
+    all_foss = ContributorRole.objects.filter(user_id=user,status=1).values('foss_category')
+    list_count = list({v['foss_category']:v for v in all_foss})
+    if len(list_count) >=3:
+        return True
+    return False
+
+def single_tutorial_allocater(request,tuto,tut,lid,days,user):  
```

Comment:
Kindly clean this function.
Use proper variable names, avoid repetition, reduce the length of the function and avoid using try...except. In case if try...except is used then mention the Exception caught.
This goes for all the below defined functions.

Reply:
I have written in such a way that , if try fails then except should get passed(considering the architecture of ST).
We can discuss the logic on a meeting.